File: backend/transactions/ward_infections/create_ward_infection_transaction.py
from ward_infections.add_ward_infection import add_ward_infection
from schemas.ward_infection_schemas import AddWardInfectionRequest
from supabase import Client


def create_ward_infection_transaction(
                                    client:Client,
                                    ward_infection: AddWardInfectionRequest,
                                    hospital_id: str
):

    print("create_ward_infection_transaction")

    return add_ward_infection(
                                client=client, 
                                ward_infection=ward_infection,
                                hospital_id=hospital_id
                            )

File: frontend/app/api/transactions/wardInfections/createWardInfectionTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { authFetch } from "../../client/apiClient"

import { CreateWardInfectionType } from "../../../types/wardInfectionTypes"

import { getWardInfectionsFromApi } from "../../wardInfections/fetchWardInfections"

import {
  normalizeWardInfection,
  toCreateWardInfectionRequest
} from "../../../utils/wardInfectionMapper"

type CreateWardInfectionTransactionParams = {
  wardInfection: CreateWardInfectionType
  setWardInfections: any
  onClose?: () => void
}

export async function createWardInfectionTransaction({
  wardInfection,
  setWardInfections,
  onClose
}: CreateWardInfectionTransactionParams)
{
  console.log("createWardInfectionTransaction")

  await fetch(
    `${API_BASE_URL}/ward-infections`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      credentials: "include",
      body: JSON.stringify(
        toCreateWardInfectionRequest(
          wardInfection
        )
      )
    }
  )

  const wardInfections =
    await getWardInfectionsFromApi()

  setWardInfections(
    wardInfections.map(
      normalizeWardInfection
    )
  )

  if (onClose) {
    onClose()
  }
}